iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/ReferenceTerminalsImpl.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2023, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl.extensions;
+
+import com.google.common.collect.ImmutableSet;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.ReferenceTerminals;
+import com.powsybl.iidm.network.impl.AbstractMultiVariantIdentifiableExtension;
+
+import java.util.*;
+
+/**
+ * @author Damien Jeandemange {@literal <damien.jeandemange at artelys.com>}
+ */
+class ReferenceTerminalsImpl extends AbstractMultiVariantIdentifiableExtension<Network> implements ReferenceTerminals {
+
+    private final class ReferenceTerminalsListener extends DefaultNetworkListener {
```

Comment:
Very interesting way to solve the terminal removal cleanup issue. I think we have a bug in SlackTerminal implementation...

Reply:
Yes SlackTerminal, also RemoteReactivePowerControl (but this one will be part of iIDM one day), maybe others, to be reviewed and fixed in other PRs.